### DISCUSSION TITLE
Help with bluetooth commands

### DISCUSSION THREAD

**Discussion by Narsku** [darthcloud/BlueRetro#290] [Q&A] [answered]
Hello!, im making a bluetooth gamepad with blegamepad library but this doesn't have vibration.
My question is what are the bluetooth commands that uses the gamepads for the vibration? or what code receive to start the vibration feature

Thanks in advance

**Comment by darthcloud** [comment, marked as answer]
Rumble is not really a standard feature and as such BlueRetro does not support rumble for generic gamepad.

What you should do is to make your gamepad to use the Xbox X|S hid descriptor and reports and this way BlueRetro will recognize it as such and you can implement rumble base on the xinput report style.

See other discussion #282 & #293
